### PR TITLE
fix navigation sidebar

### PR DESCRIPTION
**Pull request by AhmedCodeGuy**
# Summary

Fixes #576

Fixes the navigation sidebar shifting on mushaf settings changes.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Test plan

This was tested for web/mobile (chrome), and LTR/RTL.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/components/QuranReader/SidebarNavigation/SidebarNavigation.tsx [hunk 1]
```diff
@@ -70,10 +70,14 @@ const SidebarNavigation = () => {
...
-        [styles.spaceOnTopMenuExpanded]: isContextMenuExpanded,
-        [styles.spaceOnTopMenuCollapse]: !isContextMenuExpanded,
+        [styles.visibleContainer]: isSidebarVisible === true && isNavbarVisible,
+        [styles.visibleContainerCollapsed]: isSidebarVisible === true && !isNavbarVisible,
+        [styles.containerAuto]: isSidebarVisible === 'auto' && isNavbarVisible,
+        [styles.containerAutoCollapsed]: isSidebarVisible === 'auto' && !isNavbarVisible,
+        [styles.inVisibleContainer]: isNavbarVisible,
+        [styles.inVisibleContainerCollapsed]: !isNavbarVisible,
+        [styles.spaceOnTop]: isSidebarVisible && isNavbarVisible,
+        [styles.spaceOnTopCollapsed]: isSidebarVisible && !isNavbarVisible,
```
Comment (osamasayed) [inline src/components/QuranReader/SidebarNavigation/SidebarNavigation.tsx:80]: I see there are a few duplications in this section. For example:
- `isSidebarVisible === true && isNavbarVisible` and `isSidebarVisible && isNavbarVisible`
- `isSidebarVisible && !isNavbarVisible` and `isSidebarVisible === true && !isNavbarVisible`

We should keep it DRY and technically they should share the same exact classes

**Review by osamasayed — CHANGES_REQUESTED** [review]

File: src/components/QuranReader/SidebarNavigation/SidebarNavigation.tsx (hunk 1 above)
Comment (AhmedCodeGuy) [reply src/components/QuranReader/SidebarNavigation/SidebarNavigation.tsx:80]: `isSidebarVisible === true` is different from `isNavbarVisible` as `isNavbarVisible` could be `true` or `auto`, however the first condition applies only when `isNavbarVisible` equals to `true` only, same for the second point.

Comment (osamasayed) [reply src/components/QuranReader/SidebarNavigation/SidebarNavigation.tsx:80]: If this is the case, the code is very hard to follow. We defo need to refactor it with better naming and introduce variables. E.g. extract `isSidebarVisible === 'auto'` to a variable since it's used twice for example.

Comment (AhmedCodeGuy) [reply src/components/QuranReader/SidebarNavigation/SidebarNavigation.tsx:80]: `isSidebarVisible === true` is different from `isSidebarVisible` as `isSidebarVisible` could be `true` or `auto`.

File: src/components/QuranReader/SidebarNavigation/SidebarNavigation.module.scss [hunk 2]
```diff
@@ -41,15 +48,46 @@ $progress-bar-height: var(--spacing-micro);
   display: flex;
   flex-direction: column;
   overflow-y: hidden;
-  @include sidebarInvisible();
+  margin-top: -1px;
```
Comment (osamasayed) [inline src/components/QuranReader/SidebarNavigation/SidebarNavigation.module.scss:51]: If we need a negative value, we usually do `calc(-1*var...)`

**Review by osamasayed — CHANGES_REQUESTED** [review]

**Review by osamasayed — APPROVED** [review]
